components/public-api/gitpod/v1/user.proto

```diff
@@ -11,13 +11,31 @@ option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
 service UserService {
   // GetAuthenticatedUser allows to retrieve the current user.
   rpc GetAuthenticatedUser(GetAuthenticatedUserRequest) returns (GetAuthenticatedUserResponse) {}
+
+  // UpdateUser updates the properties of a user.
+  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse) {}
 }
 
 message GetAuthenticatedUserRequest {}
 message GetAuthenticatedUserResponse {
   User user = 1;
 }
 
+message UpdateUserRequest {
+  string user_id = 1;
+  optional string name = 2;
+  optional string accepted_privacy_policy_date = 3;
+  optional ProfileDetails profile = 4;
```

Comment:
we should also mark optional internal shapes and only declared updatable properties, please see Update of ConfigurationService

We should not share read shapes.

Reply:
Done.